twill-yarn/src/main/java/org/apache/twill/filesystem/FileContextLocation.java

```diff
@@ -162,7 +163,8 @@ public URI toURI() {
     // append "port" to the path URI, while the DistributedFileSystem always use the cluster logical
     // name, which doesn't allow having port in it.
     URI uri = path.toUri();
-    if (HAUtil.isLogicalUri(locationFactory.getConfiguration(), uri)) {
+
+    if (FileContextLocationUtil.useLogicalUri(locationFactory.getConfiguration(), uri)) {
```

Comment:
I am not sure it is as simple as that. We may need to change equals() to compare based on toURI() and not path or somehow make path and toURI() to be in sync within that class, otherwise parts become inconsistent and I am not sure unitTests failures in this case is the only issue.